staging/src/k8s.io/apimachinery/pkg/labels/labels.go

```diff
@@ -71,6 +71,15 @@ func (ls Set) AsSelectorPreValidated() Selector {
 	return SelectorFromValidatedSet(ls)
 }
 
+// AsSelectorPreValidated converts labels into a selector, but
+// assumes that labels are already validated and thus don't
+// preform any validation.
+// According to our measurements this is significantly faster
+// in codepaths that matter at high scale.
```

Comment:
given the number of allocations this still uses, I would expect any code path that cares about performance at high scale to avoid repeated construction of identical selectors

Reply:
Maybe we can have map[Set]Selector - if the lookup of Selector doesn't find it, we construct a new Selector.

@liggitt 
What do you think ?